planner/core/logical_plan_builder.go

```diff
@@ -2216,7 +2245,7 @@ func getStatsTable(ctx sessionctx.Context, tblInfo *model.TableInfo, pid int64)
 	return statsTbl
 }
 
-func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName) (LogicalPlan, error) {
+func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, asName model.CIStr) (LogicalPlan, error) {
```

Comment:
You could try not to push the `asName` when `buildDataSource`. There is only a place calling `buildDataSource`. The alias will be set after calling `buildDataSource`.

Reply:
what do you mean? we need to distinguish `t1`, `t2` and `t` in this case: `from t t1, t t2`.

Reply:
I mean the alias of `DataSource` is set after `buildDataSource`. We can handle hints after that rather than do it here.

Reply:
Original SQL index hint like `select * from t t1 use index(idx)` is handled here, so I handle comment-style index hint here, too. the original one don't need alias because `use index` must follow `t t1`, we don't need know the alias for this. But in comment-style hint, we need to write `/*+ INDEX(t1, idx) */` or `/*+ INDEX(t, idx) */`, these two are different, you can take a look at test case.

Reply:
I didn't mean this...

Reply:
I didn't mean that.

Reply:
I didn't advise you for not considering the `asName`. What I suggest is to move the hints handle after this function. Because the alias of the `DataSource` is set after the function.

Reply:
You can take a look at https://github.com/pingcap/tidb/blob/096deba00b09708b844fdc65c042ad5a6d05dbf9/planner/core/logical_plan_builder.go#L176